src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -19,21 +19,30 @@ public DatabaseTreeNode(ServerNode serverNode, string databaseName): this()
         {
             Parent = serverNode;
             NodeValue = databaseName;
-            Database db = new Database(serverNode.GetContextAs<SmoQueryContext>().Server, this.NodeValue);
+            var db = new Database(serverNode.GetContextAs<SmoQueryContext>().Server, this.NodeValue);
             db.Refresh();
+            // If we got this far, the connection is valid. However, it's possible
+            // the user connected directly to the master of a readable secondary
+            // In that case, the name in the connection string won't be found in sys.databases
+            // We detect that here and fall back to master
```

Comment:
Might be worth mentioning that we "detect" that by the fact that SMO will leave the DB as "Creating"

Reply:
I don't see that the code suggests we're detecting it any other way

Reply:
Yeah it's probably fine but I'm not sure someone not familiar with SMO would look at this and not be confused why the state would even be "Creating".